File: frontEnd/src/app/shared/model/produto.ts
// export class Produto {
//   constructor(
//   public id: number,
//   public nome: string,
//   public dataCompra: Date,
//   public duracaoGarantiaMeses: number,
//   public dataFimGarantia: string,
//   )
//   {}
// }

export class Produto {
  id?: string;
  nome: string;
  dataCompra: Date;
  duracaoGarantiaMeses: number;
  dataFimGarantia?: Date;

  constructor(nome: string, dataCompra: Date, duracaoGarantiaMeses: number, dataFimGarantia: Date) {
    this.nome = nome;
    this.dataCompra = dataCompra;
    this.duracaoGarantiaMeses = duracaoGarantiaMeses;
    this.dataFimGarantia = dataFimGarantia;
  }
}

File: backEnd/src/main/java/com/garantia/api/model/Produto.java
package com.garantia.api.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Data
@Entity
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 200, nullable = false)
    private String nome;

//    @Column(length = 200, nullable = false)
    private Date dataCompra;

//    @Column(length = 200, nullable = false)
    private Integer duracaoGarantiaMeses;

//    @Column(length = 200, nullable = false)
    private Date dataFimGarantia;
}
